### PR TITLE
rpc/wallet: Add coinType to queryOptions of listunspent

### PR DESCRIPTION
**Pull request by xdustinface**
This PR allows to filter utxos by `CoinType` when querying them with the `listunspent` rpc command like 

```
listunspent 1 99999 "[]" true "{\"coinType\":3}"
```

to query only non-denominated utxos. 

It also renames the CoinType `ONLY_1000` to `ONLY_MASTERNODE_COLLATERAL` to make it more obvious in the code and the rpc description what that's for.. hope thats ok!

### CODE REVIEW

File: src/wallet/rpcwallet.cpp [hunk 1]
```diff
@@ -3081,6 +3084,18 @@ UniValue listunspent(const JSONRPCRequest& request)
...
+
+            int64_t nCoinType = options["coinType"].get_int64();
+
+            if (nCoinType > static_cast<int64_t>(CoinType::MAX_COIN_TYPE)) {
+                throw JSONRPCError(RPC_INVALID_PARAMETER, strprintf("Invalid coinType selected. Available range: %d - %d", static_cast<int64_t>(CoinType::ALL_COINS), static_cast<int64_t>(CoinType::MAX_COIN_TYPE)));
+            }
+
+            coinControl.nCoinType = static_cast<CoinType>(nCoinType);
+        }
+
```
Comment (PastaPastaPasta) [inline src/wallet/rpcwallet.cpp:3098]: nit: I would drop most if not all of these empty lines, but that's mostly preference

File: src/wallet/rpcwallet.cpp [hunk 2]
```diff
@@ -2996,6 +2996,7 @@ UniValue listunspent(const JSONRPCRequest& request)
             "      \"maximumAmount\"    (numeric or string, default=unlimited) Maximum value of each UTXO in " + CURRENCY_UNIT + "\n"
             "      \"maximumCount\"     (numeric or string, default=unlimited) Maximum number of UTXOs\n"
             "      \"minimumSumAmount\" (numeric or string, default=unlimited) Minimum sum value of all UTXOs in " + CURRENCY_UNIT + "\n"
+            "      \"coinType\"         (numeric, default=0) Filter coinTypes as followed: 0=ALL_COINS, 1=ONLY_FULLY_MIXED, 2=ONLY_READY_TO_MIX, 3=ONLY_NONDENOMINATED, 4=ONLY_MASTERNODE_COLLATERAL, 5=ONLY_PRIVATESEND_COLLATERAL\n"
```
Comment (UdjinM6) [inline src/wallet/rpcwallet.cpp:2999]:
```suggestion
            "      \"coinType\"         (numeric, default=0) Filter coinTypes as follows:\n"
            "                         0=ALL_COINS, 1=ONLY_FULLY_MIXED, 2=ONLY_READY_TO_MIX, 3=ONLY_NONDENOMINATED,\n"
            "                         4=ONLY_MASTERNODE_COLLATERAL, 5=ONLY_PRIVATESEND_COLLATERAL\n"
```

File: src/wallet/rpcwallet.cpp (hunk 1 above)
Comment (xdustinface) [reply src/wallet/rpcwallet.cpp]: can you live with half of them like in ef2d027? 😅

File: src/wallet/rpcwallet.cpp [hunk 3]
```diff
@@ -3090,7 +3090,7 @@ UniValue listunspent(const JSONRPCRequest& request)
         if (options.exists("coinType")) {
             int64_t nCoinType = options["coinType"].get_int64();
 
-            if (nCoinType > static_cast<int64_t>(CoinType::MAX_COIN_TYPE)) {
+            if (nCoinType < static_cast<int64_t>(CoinType::ALL_COINS) || nCoinType > static_cast<int64_t>(CoinType::MAX_COIN_TYPE)) {
```
Comment (UdjinM6) [inline src/wallet/rpcwallet.cpp:3093]: Good catch! 👍 But I would rather prefer introducing `MIN_COIN_TYPE = ALL_COINS,` and using it here and in the error message below.

Comment (xdustinface) [reply src/wallet/rpcwallet.cpp]: Makes sense 👍 changed!

**Comment by PastaPastaPasta** [comment]
@xdustinface if you find time/ you can do this in another PR, having some functional tests for this RPC command / for this specific feature of the command would be nice

File: src/wallet/rpcwallet.cpp [hunk 4]
```diff
@@ -3066,6 +3069,8 @@ UniValue listunspent(const JSONRPCRequest& request)
     CAmount nMaximumAmount = MAX_MONEY;
     CAmount nMinimumSumAmount = MAX_MONEY;
     uint64_t nMaximumCount = 0;
+    CCoinControl coinControl;
```
Comment (10xcryptodev) [inline src/wallet/rpcwallet.cpp:3072]: nit: most of the code base use coin_control as variable name, i think it's good use the same pattern. Not sure if there is a variable name standard for the codebase

Comment (PastaPastaPasta) [reply src/wallet/rpcwallet.cpp:3072]: Hmm, it doesn't look like the code-base has a standard, looking at the whole codebase, both are used quite actively, and in rpcwallet.cpp both are used.

**Pull request merged**